### DISCUSSION TITLE
Upgrade sonoff zbbridge firmware fails.

### DISCUSSION THREAD

**Discussion by shyney7** [arendst/Tasmota#16103] [Q&A] [answered]
Since the newest release of home assistant my zigbee devices stopped working I don't think that this is a tasmota fw issue but I want to try everything before switching to another device. My current firmware is **9.2.0** and the ota upgrade doesn't seem to work.

```
00:25:35 CMD: upgrade 1
00:25:35 RSL: stat/tasmota_zbbridge/RESULT = {"Upgrade":"Version 9.2.0 from http://ota.tasmota.com/tasmota/tasmota-zbbridge.bin.gz"}
00:25:38 HTP: Web server stopped
00:25:53 HTP: Web server active on sonoff-zbbridge with IP address 192.168.1.85
00:25:54 HTP: Web server stopped
00:26:09 HTP: Web server active on sonoff-zbbridge with IP address 192.168.1.85
00:26:10 HTP: Web server stopped
00:26:25 HTP: Web server active on sonoff-zbbridge with IP address 192.168.1.85
00:26:26 HTP: Web server stopped
00:26:41 HTP: Web server active on sonoff-zbbridge with IP address 192.168.1.85
00:26:42 HTP: Web server stopped
00:26:43 HTP: Web server active on sonoff-zbbridge with IP address 192.168.1.85
00:26:44 RSL: stat/tasmota_zbbridge/UPGRADE = {"Upgrade":"Failed HTTP error: connection failed"}
```

It doesn't matter if I use http://ota.tasmota.com/tasmota/tasmota-zbbridge.bin.gz 
or
http://ota.tasmota.com/tasmota/release/tasmota-zbbridge.bin.gz

And if I try to upgrade via file upload I get `Upload buffer miscompare`

My zbbridge is hot glued inside a closet so I would rather upgrade without the need of taking it out and using a flasher device.

**Comment by s-hadinger** [comment, marked as answer]
Looks like a dns or internet issue. You can always download the firmware on your local computer and upload it directly in the web ui

**Reply by shyney7** [reply]
> And if I try to upgrade via file upload I get Upload buffer miscompare

**Reply by s-hadinger** [reply]
Then upload tasmota-minimal first, and then tasmota-zbbridge.

Just a piece of warning though, the ser2net part used by ZHA hasn't change since v9.2.0. So don't have too much expectations of success

**Reply by shyney7** [reply]
@s-hadinger oh thx for that info. I've read from people with similar issues that switching from ZHA to Zigbee2MQTT solved all problems. Do you know if there is a way of migrating all paired devices to Zigbee2MQTT without needing to pair them all again?
